android/tangram/src/com/mapzen/tangram/MapController.java

```diff
@@ -179,11 +176,22 @@ void init() {
         // Parse font file desription
         fontFileParser.parse();
 
-        nativeInit(this, assetManager, scenePath);
+        mapPointer = nativeInit(this, assetManager);
+    }
+
+    void dispose() {
+        // Disposing native resources involves GL calls, so we need to run on the GL thread.
+        queueEvent(new Runnable() {
+            @Override
+            public void run() {
+                nativeDispose(mapPointer);
+                mapPointer = 0;
```

Comment:
Should we not be clearing `clientDataSources` for this Map instance, at this point?